tests/pytests/unit/states/test_gem.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+import salt.states.gem as gem
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture
+def configure_loader_modules():
+    return {gem: {"__opts__": {"test": False}}}
+
+
+def test_installed():
+    gems = {"foo": ["1.0"], "bar": ["2.0"]}
+    gem_list = MagicMock(return_value=gems)
+    gem_install_succeeds = MagicMock(return_value=True)
+    gem_install_fails = MagicMock(return_value=False)
+
+    with patch.dict(gem.__salt__, {"gem.list": gem_list}):
+        with patch.dict(gem.__salt__, {"gem.install": gem_install_succeeds}):
```

Comment:
**suggestion (non-blocking)** these could be one `with` statement using side-effect

```
with patch.dict(..., {"gem.list": gem_list, "gem_install": MagicMock(side_effect=[True, False])}):
    ...
```

Side effect as an iterable will return the items in order with each call.

Reply:
(if this were changed, the assert should probably be changed to `assert_has_calls([call(...), call(...)])` (success then fail call, which at a glance looks to be two identical calls?)

Reply:
If this change is made, should be made throughout